The warnings should not fail in the CLI

I expect the warnings should not fail in the CLI, just the errors.

<img width="799" alt="Screenshot 2020-04-28 at 20 03 09" src="https://user-images.githubusercontent.com/1007051/80521482-62d7f100-898b-11ea-9b08-8a849a6bbf97.png">

```
> lit-analyzer "src/{components,pages}/**/*.ts" --strict

Analyzing 4 files...

./src/components/app-shell.ts

    Unknown tag <pwa-install-button>.
    67:  <pwa-install-button>
    no-unknown-tag-name

    Unknown tag <pwa-update-available>.
    71:  <pwa-update-available>
    no-unknown-tag-name

  ✖ 2 problems in 1 file (0 errors, 2 warnings)

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! pwa-starter@0.1.0 lit:check-types: `lit-analyzer "src/{components,pages}/**/*.ts" --strict`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the pwa-starter@0.1.0 lit:check-types script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

An example with ESLint:

<img width="617" alt="Screenshot 2020-04-28 at 20 06 32" src="https://user-images.githubusercontent.com/1007051/80521811-ee518200-898b-11ea-91b9-d4809ab0a780.png">

--- a/packages/lit-analyzer/README.md
+++ b/packages/lit-analyzer/README.md
@@ -38,7 +38,7 @@ npm install lit-analyzer -g
 lit-analyzer src
 lit-analyzer "src/**/*.{js,ts}"
 lit-analyzer my-element.js
-lit-analyzer --format markdown --outFile result.md 
+lit-analyzer --format markdown --outFile result.md
 ```
 
 [![-----------------------------------------------------](https://raw.githubusercontent.com/andreasbm/readme/master/assets/lines/rainbow.png)](#configuration)
@@ -63,7 +63,7 @@ lit-analyzer --strict --rules.no-unknown-tag-name off --format markdown
 | `--rules.rule-name` | Enable or disable rules (example: --rules.no-unknown-tag-name off). Severity can be "off" \| "warn" \| "error". See a list of rules [here](https://github.com/runem/lit-analyzer/blob/master/docs/readme/rules.md). | `{"rule-name": "off" \| "warn" \| "error"}` |  |
 | `--strict` | Enable strict mode. This changes the default ruleset | `boolean` | |
 | `--format` | Change the format of how diagnostics are reported | `code` \| `list` \| `markdown` | code |
-| `--maxWarnings` | Fail only when the number of warnings is larger than this number | `number` | 0 |
+| `--maxWarnings` | Fail only when the number of warnings is larger than this number | `number` | -1 |
 | `--outFile` | Emit all output to a single file  | `filePath` |  |
 | `--quiet` | Report only errors and not warnings | `boolean` |  |
 | `--failFast` | Exit the process right after the first problem has been found | `boolean` | |
@@ -633,7 +633,7 @@ Code is analyzed using [web-component-analyzer](https://github.com/runem/web-com
  * @slot right - Right content
  * @slot left
  */
-class MyElement extends HTMLElement { 
+class MyElement extends HTMLElement {
 }
 
 customElements.define("my-element", MyElement);

--- a/packages/lit-analyzer/readme/config.md
+++ b/packages/lit-analyzer/readme/config.md
@@ -18,7 +18,7 @@ lit-analyzer --strict --rules.no-unknown-tag-name off --format markdown
 | `--rules.rule-name` | Enable or disable rules (example: --rules.no-unknown-tag-name off). Severity can be "off" \| "warn" \| "error". See a list of rules [here](https://github.com/runem/lit-analyzer/blob/master/docs/readme/rules.md). | `{"rule-name": "off" \| "warn" \| "error"}` |  |
 | `--strict` | Enable strict mode. This changes the default ruleset | `boolean` | |
 | `--format` | Change the format of how diagnostics are reported | `code` \| `list` \| `markdown` | code |
-| `--maxWarnings` | Fail only when the number of warnings is larger than this number | `number` | 0 |
+| `--maxWarnings` | Fail only when the number of warnings is larger than this number | `number` | -1 |
 | `--outFile` | Emit all output to a single file  | `filePath` |  |
 | `--quiet` | Report only errors and not warnings | `boolean` |  |
 | `--failFast` | Exit the process right after the first problem has been found | `boolean` | |

--- a/packages/lit-analyzer/src/analyze/parse/convert-component-definitions-to-html-collection.ts
+++ b/packages/lit-analyzer/src/analyze/parse/convert-component-definitions-to-html-collection.ts
@@ -94,7 +94,7 @@ export function convertComponentDeclarationToHtmlTag(
 				required: member.required
 			});
 
-			if (!("attrName" in member) && addDeclarationPropertiesAsAttributes && (definition != null && !definition.fromLib)) {
+			if (!("attrName" in member) && addDeclarationPropertiesAsAttributes && definition != null && !definition.fromLib) {
 				if (declaration.node.getSourceFile().isDeclarationFile) {
 					htmlTag.attributes.push({
 						...base,

--- a/packages/lit-analyzer/src/analyze/parse/document/virtual-document/virtual-ast-document.ts
+++ b/packages/lit-analyzer/src/analyze/parse/document/virtual-document/virtual-ast-document.ts
@@ -65,7 +65,8 @@ export class VirtualAstDocument implements VirtualDocument {
 
 				if (
 					(range.start < literalPartRange.start && range.end > literalPartRange.end) ||
-					(intersects(range.start + 1, literalPartRange) || intersects(range.end - 1, literalPartRange))
+					intersects(range.start + 1, literalPartRange) ||
+					intersects(range.end - 1, literalPartRange)
 				) {
 					const strStart = Math.max(literalPartRange.start, range.start);
 					const strEnd = Math.min(literalPartRange.end, range.end);

--- a/packages/lit-analyzer/src/analyze/util/type/is-assignable-binding-under-security-system.ts
+++ b/packages/lit-analyzer/src/analyze/util/type/is-assignable-binding-under-security-system.ts
@@ -56,7 +56,7 @@ const closureScopedOverrides: TagNameToSecurityOverrideMap = {
 	},
 	source: {
 		src: ["TrustedResourceUrl", "SafeUrl"]
-	},
+	}
 };
 const closureGlobalOverrides: SecurityOverrideMap = {
 	style: ["SafeStyle", "string"]

--- a/packages/lit-analyzer/src/cli/analyze-command.ts
+++ b/packages/lit-analyzer/src/cli/analyze-command.ts
@@ -111,7 +111,11 @@ export async function analyzeCommand(globs: string[], config: LitAnalyzerCliConf
 		printText(statsText, config);
 	}
 
-	return stats.diagnostics <= (config.maxWarnings || 0) || config.maxWarnings === -1;
+	if (stats.errors !== 0) {
+		return false;
+	}
+
+	return stats.warnings <= (config.maxWarnings || 0) || config.maxWarnings === -1;
 }
 
 function getFormatter(format: FormatterFormat): DiagnosticFormatter {

--- a/packages/lit-analyzer/src/cli/cli.ts
+++ b/packages/lit-analyzer/src/cli/cli.ts
@@ -33,13 +33,10 @@ export async function cli() {
 	}
 
 	// Always convert "rules" to "dash case" because "rules" expects it.
-	config.rules = Object.entries(config.rules || {}).reduce(
-		(acc, [k, v]) => {
-			acc[camelToDashCase(k) as LitAnalyzerRuleName] = v;
-			return acc;
-		},
-		{} as LitAnalyzerRules
-	);
+	config.rules = Object.entries(config.rules || {}).reduce((acc, [k, v]) => {
+		acc[camelToDashCase(k) as LitAnalyzerRuleName] = v;
+		return acc;
+	}, {} as LitAnalyzerRules);
 
 	if (config.help) {
 		// eslint-disable-next-line no-console

--- a/packages/lit-analyzer/src/cli/cli.ts
+++ b/packages/lit-analyzer/src/cli/cli.ts
@@ -9,7 +9,7 @@ const DEFAULT_GLOB = "src/**/*.{js,jsx,ts,tsx}";
 const DEFAULT_CONFIG: LitAnalyzerCliConfig = {
 	noColor: false,
 	quiet: false,
-	maxWarnings: 0,
+	maxWarnings: -1,
 	debug: false,
 	help: false,
 	failFast: false,

--- a/packages/lit-analyzer/src/cli/format/util.ts
+++ b/packages/lit-analyzer/src/cli/format/util.ts
@@ -9,12 +9,16 @@ export function generalReport(stats: AnalysisStats): string {
 	}
 
 	if (stats.diagnostics > 0) {
-		return `\n${chalk.red(
-			`  ✖ ${numberStatText(stats.diagnostics, "problem")} in ${numberStatText(stats.filesWithProblems, "file")} (${numberStatText(
-				stats.errors,
-				"error"
-			)}, ${numberStatText(stats.warnings, "warning")})`
-		)}`;
+		const message = `  ✖ ${numberStatText(stats.diagnostics, "problem")} in ${numberStatText(stats.filesWithProblems, "file")} (${numberStatText(
+			stats.errors,
+			"error"
+		)}, ${numberStatText(stats.warnings, "warning")})`;
+
+		if (stats.errors > 0) {
+			return `\n${chalk.red(message)}`;
+		} else {
+			return `\n${chalk.yellow(message)}`;
+		}
 	} else {
 		return `\n${chalk.green(`  ✓ Found 0 problems in ${numberStatText(stats.totalFiles, "file")}`)}`;
 	}

--- a/packages/lit-analyzer/src/rules/no-nullable-attribute-binding.ts
+++ b/packages/lit-analyzer/src/rules/no-nullable-attribute-binding.ts
@@ -31,7 +31,7 @@ const rule: RuleModule = {
 					source: "no-nullable-attribute-binding",
 					severity: litDiagnosticRuleSeverity(request.config, "no-nullable-attribute-binding"),
 					location: { document: request.document, ...htmlAttr.location.name },
-					htmlAttr: htmlAttr as typeof htmlAttr & ({ assignment: typeof assignment }),
+					htmlAttr: htmlAttr as typeof htmlAttr & { assignment: typeof assignment },
 					typeA,
 					typeB
 				}
@@ -48,7 +48,7 @@ const rule: RuleModule = {
 					source: "no-nullable-attribute-binding",
 					severity: litDiagnosticRuleSeverity(request.config, "no-nullable-attribute-binding"),
 					location: { document: request.document, ...htmlAttr.location.name },
-					htmlAttr: htmlAttr as typeof htmlAttr & ({ assignment: typeof assignment }),
+					htmlAttr: htmlAttr as typeof htmlAttr & { assignment: typeof assignment },
 					typeA,
 					typeB
 				}

--- a/packages/lit-analyzer/src/rules/no-unknown-tag-name.ts
+++ b/packages/lit-analyzer/src/rules/no-unknown-tag-name.ts
@@ -19,7 +19,10 @@ const rule: RuleModule = {
 		// Add diagnostics if the tag couldn't be found (not defined anywhere)
 		if (htmlTag == null) {
 			// Find a suggested name in the set of defined tag names. Maybe this tag name is a typo?
-			const suggestedName = findBestStringMatch(htmlNode.tagName, Array.from(htmlStore.getGlobalTags()).map(tag => tag.tagName));
+			const suggestedName = findBestStringMatch(
+				htmlNode.tagName,
+				Array.from(htmlStore.getGlobalTags()).map(tag => tag.tagName)
+			);
 
 			// Build a suggestion text
 			let suggestion = `Check that you've imported the element, and that it's declared on the HTMLElementTagNameMap.`;

--- a/packages/lit-analyzer/test/rules/no-unknown-tag-name.ts
+++ b/packages/lit-analyzer/test/rules/no-unknown-tag-name.ts
@@ -26,7 +26,9 @@ test("Don't report known custom elements found in other file", t => {
 test("Don't report known custom element", t => {
 	const { diagnostics } = getDiagnostics(
 		"class MyElement extends HTMLElement {}; customElements.define('my-element', MyElement); html`<my-element></my-element>`",
-		{ rules: { "no-unknown-tag-name": true } }
+		{
+			rules: { "no-unknown-tag-name": true }
+		}
 	);
 	hasNoDiagnostics(t, diagnostics);
 });

--- a/packages/vscode-lit-plugin/src/extension.ts
+++ b/packages/vscode-lit-plugin/src/extension.ts
@@ -45,7 +45,10 @@ export async function activate(context: vscode.ExtensionContext) {
 
 	// Register a color provider
 	const registration = vscode.languages.registerColorProvider(
-		[{ scheme: "file", language: "typescript" }, { scheme: "file", language: "javascript" }],
+		[
+			{ scheme: "file", language: "typescript" },
+			{ scheme: "file", language: "javascript" }
+		],
 		colorProvider
 	);
 	context.subscriptions.push(registration);

